File: app/src/main/java/com/omar/retromp3recorder/app/usecase/ChangeSampleRateUC.java
package com.omar.retromp3recorder.app.usecase;

import com.omar.retromp3recorder.app.recorder.VoiceRecorder;
import com.omar.retromp3recorder.app.repo.SampleRateRepo;

import javax.inject.Inject;

import io.reactivex.Completable;

public class ChangeSampleRateUC {
    private final SampleRateRepo repo;

    @Inject
    public ChangeSampleRateUC(SampleRateRepo repo) {
        this.repo = repo;
    }

    public Completable execute(VoiceRecorder.SampleRate sampleRate) {
        return Completable.fromAction(() -> repo.newValue(sampleRate));
    }
}

File: app/src/main/java/com/omar/retromp3recorder/app/usecases/ChangeSampleRateUC.kt
package com.omar.retromp3recorder.app.usecases

import com.omar.retromp3recorder.app.state.SampleRateRepo
import com.omar.retromp3recorder.recorder.Mp3VoiceRecorder
import io.reactivex.Completable
import javax.inject.Inject

class ChangeSampleRateUC @Inject constructor(private val repo: SampleRateRepo) {
    fun execute(sampleRate: Mp3VoiceRecorder.SampleRate): Completable {
        return Completable.fromAction { repo.newValue(sampleRate) }
    }
}
